### PR TITLE
feat(veri): implement verifreg diff methods and update shims

### PR DESCRIPTION
**Pull request by frrist**
- define task for tracking verifreg actor
- define actor shim with codegen and diffing logic for verifreg actor
- define sql migration for new task tables

### CODE REVIEW

File: schemas/v1/schema.go [hunk 1]
```diff
@@ -1366,3 +1371,63 @@ CREATE INDEX state_heights_height_index ON {{ .SchemaName | default "public"}}.s
...
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64, reflect.Uintptr:
+		return v.Uint() == 0
+	case reflect.Float32, reflect.Float64:
+		return v.Float() == 0
+	case reflect.Struct:
+		return false
+	default:
+		return v.IsNil()
+	}
+}
```
Comment (frrist) [inline schemas/v1/schema.go:1433]: I took the easy way out and copy-pasted some lines from the storage model to get this to create valid migrations for the new models. @iand how should we be specifying Schema names in the migration?

File: tasks/messages/types/ipldsch_minima.go [hunk 2]
```diff
@@ -10,14 +10,14 @@ import (
 )
```
Comment (frrist) [inline tasks/messages/types/ipldsch_minima.go:10]: gofmt'd these files and the others in this dif.

**Review by iand — CHANGES_REQUESTED** [review]

File: tasks/messages/types/ipldsch_minima.go (hunk 2 above)
Comment (iand) [reply tasks/messages/types/ipldsch_minima.go:10]: We should add that as a step in the generation task

File: schemas/v1/schema.go (hunk 1 above)
Comment (iand) [reply schemas/v1/schema.go:1433]: See https://github.com/filecoin-project/sentinel-visor/pull/578 for migration capability

File: tasks/actorstate/verifreg.go [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+		}
+		return verifiers, events, nil
+	}
+
+	changes, err := verifreg.DiffVerifiers(ctx, ec.Store, ec.PrevState, ec.CurrState)
+	if err != nil {
+		return nil, nil, xerrors.Errorf("diffing verifier registry verifiers: %w", err)
+	}
+
+	// a new verifier was added
```
Comment (iand) [inline tasks/actorstate/verifreg.go:143]: Do we need separate a separate VerifiedRegistryVerifierEvent model? Can we add the Event field to the VerifiedRegistryVerifier model instead? Add a `MODIFIED` event to represent the datacap being changed.

````Go
type VerifiedRegistryVerifier struct {
	Height    int64  `pg:",pk,notnull,use_zero"`
	StateRoot string `pg:",pk,notnull"`
	Address   string `pg:",pk,notnull"`
        Event string `pg:",pk,notnull,type:verified_registry_event_type"`
	DataCap string `pg:"type:numeric,notnull"`
}
```

When Event = REMOVED then the DataCap field is zero.

File: tasks/actorstate/verifreg.go [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+		}
+		return clients, events, nil
+	}
+
+	changes, err := verifreg.DiffVerifiedClients(ctx, ec.Store, ec.PrevState, ec.CurrState)
+	if err != nil {
+		return nil, nil, xerrors.Errorf("diffing verifier registry clients: %w", err)
+	}
+
+	for _, change := range changes.Added {
```
Comment (iand) [inline tasks/actorstate/verifreg.go:209]: Same comment about merging the Event model into the VerifiedRegistryVerifiedClient model.

File: tasks/actorstate/verifreg.go (hunk 3 above)
Comment (frrist) [reply tasks/actorstate/verifreg.go:128]: Good idea -- done.

File: tasks/actorstate/verifreg.go [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+	"github.com/filecoin-project/sentinel-visor/chain/actors/adt"
+	"github.com/filecoin-project/sentinel-visor/chain/actors/builtin/verifreg"
+	"github.com/filecoin-project/sentinel-visor/metrics"
+	"github.com/filecoin-project/sentinel-visor/model"
+)
+
+type VerifiedRegistryExtractor struct{}
+
+func init() {
+	Register(sa0builtin.VerifiedRegistryActorCodeID, VerifiedRegistryExtractor{})
```
[the remark this answers is not in the document]
Comment (frrist) [reply tasks/actorstate/verifreg.go]: Rebased, and done.

File: tasks/actorstate/verifreg.go [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+		CurrTs:    a.TipSet,
+		Store:     node.Store(),
+	}, nil
+}
+
+func (VerifiedRegistryExtractor) Extract(ctx context.Context, a ActorInfo, node ActorStateAPI) (model.Persistable, error) {
+	ctx, span := global.Tracer("").Start(ctx, "VerifiedRegistryExtractor")
+	defer span.End()
+
+	stop := metrics.Timer(ctx, metrics.ProcessingDuration)
```
Comment (iand) [inline tasks/actorstate/verifreg.go:80]:
```suggestion
	stop := metrics.Timer(ctx, metrics.StateExtractionDuration)
```

**Review by iand — APPROVED** [review]
Note metric suggestion based on change I merged earlier

**Pull request merged**